src/devices/bus/rs232/cricket.cpp

```diff
@@ -0,0 +1,244 @@
+    u8 m_data;
+
+    TIMER_DEVICE_CALLBACK_MEMBER(clock_interrupt);
+};
+
+cricket_device::cricket_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, SERIAL_CRICKET, tag, owner, clock)
+	, device_rs232_port_interface(mconfig, *this)
+	, m_mcu(*this, "mcu")
+	, m_ay1(*this, "ay1")
+	, m_ay2(*this, "ay2")
+    , m_tms(*this, "tms")
+{
+}
+
+void cricket_device::device_add_mconfig(machine_config &config)
+{
+	M6801(config, m_mcu, 4.9152_MHz_XTAL);  // Crystal verified from visual inspection
+	m_mcu->out_p1_cb().set(FUNC(cricket_device::p1_w));
+	m_mcu->in_p2_cb().set(FUNC(cricket_device::p2_r));
+	m_mcu->in_p3_cb().set(FUNC(cricket_device::p3_r));
+	m_mcu->out_p3_cb().set(FUNC(cricket_device::p3_w));
+	m_mcu->in_p4_cb().set(FUNC(cricket_device::p4_r));
+	m_mcu->out_p4_cb().set(FUNC(cricket_device::p4_w));
+	m_mcu->out_ser_tx_cb().set(FUNC(cricket_device::ser_tx_w));
+
+	SPEAKER(config, "lspeaker").front_left();
+	SPEAKER(config, "rspeaker").front_right();
+	AY8913(config, m_ay1, 4.9152_MHz_XTAL / 4);
+	m_ay1->add_route(ALL_OUTPUTS, "lspeaker", 0.5);
+	AY8913(config, m_ay2, 4.9152_MHz_XTAL / 4);
+	m_ay2->add_route(ALL_OUTPUTS, "rspeaker", 0.5);
+    TMS5220C(config, m_tms, 640000);
+	m_tms->add_route(ALL_OUTPUTS, "lspeaker", 0.5);
+	m_tms->add_route(ALL_OUTPUTS, "rspeaker", 0.5);
+
+    TIMER(config, "clock_interrupt").configure_periodic(FUNC(cricket_device::clock_interrupt), attotime::from_hz(120));
+}
+
+const tiny_rom_entry *cricket_device::device_rom_region() const
+{
+	return ROM_NAME(cricket);
+}
+
+void cricket_device::device_start()
+{
+}
+
+void cricket_device::device_reset()
+{
+    output_dcd(0);
+	output_cts(0);
+}
+
+void cricket_device::input_txd(int state)
+{
+	m_rx_state = (state & 1);
+}
+
+void cricket_device::input_dtr(int state)
+{
+	m_dtr_state = (state & 1);
+}
+
+void cricket_device::p1_w(u8 data)
+{
+    if (!BIT(data, 0))
+	{
+		m_ay1->reset_w();
+		m_ay2->reset_w();
+	}
+    // TODO: Front panel LED driven from bit 1. Flashes if clock is not setup 
+}
+
+u8 cricket_device::p3_r()
+{
+    return m_dtr_state << 6;
+}
+
+void cricket_device::p3_w(u8 data)
+{
+	if (!BIT(data, 5)) //AY1 *CS
+	{
+        if (BIT(data,2)) // BDIR
+        {
+            if (BIT(data,3)) // BC1
+                m_ay1->address_w(m_data);
+            else
+                m_ay1->data_w(m_data);
+        }
+        else if (!BIT(data,3))
+            m_data = m_ay1->data_r();
+    }
+    
+    if (!BIT(data, 4)) // AY2 *CS
+    {
+        if (BIT(data,2)) // BDIR
+        {
+            if (BIT(data,3)) // BC1
+                m_ay2->address_w(m_data);
+            else
+                m_ay2->data_w(m_data);
+        }
+        else if(!BIT(data,3))
+            m_data = m_ay2->data_r();
+    }
+    
+    if (!BIT(data, 0)) //5220 read
+        m_data = m_tms->status_r();
+    
+    if (!BIT(data, 1)) //5220 write
+        m_tms->data_w(m_data);
+
+    output_dsr(!BIT(data, 7));
+}
```

Comment:
Should this be checking for edges on bits 5, 4, 1 and 0?  Also, remember the spaces after commas, and think about what to do if multiple devices are read simultaneously.